Assets/BowAndArrowImproved/Scripts/Traps/TrapActivation.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using Unity.XR.CoreUtils;
+using UnityEngine;
+
+public class TrapActivation : MonoBehaviour
+{
+    [SerializeField] private float activationTime = 3f;
+    [SerializeField] private List<ContinuousTrap> myTraps;
+    [SerializeField] private LayerMask hitMask;
+    [SerializeField] private bool areTrapsActive;
+
+    private void OnTriggerEnter(Collider other)
+    {
+        if (!hitMask.Contains(other.attachedRigidbody.gameObject.layer) || areTrapsActive) return;
+
+        areTrapsActive = true;
+        foreach (ContinuousTrap trap in myTraps)
+        {
+            trap.ToggleTrapActivation(true);
+        }
```

Comment:
first activate the traps (foreach...), THEN set the boolean (areTrapsActive = ...)